process/transactionProcessor.go

```diff
@@ -500,24 +500,48 @@ func checkIfMoveBalanceNotarized(tx *transaction.ApiTransactionResult) bool {
 	return true
 }
 
-func (tp *TransactionProcessor) checkIfIntraShardRelayedTransaction(
-	tx *transaction.ApiTransactionResult,
-	senderAddress []byte,
-	receiverAddress []byte,
-) bool {
+func (tp *TransactionProcessor) checkIfCompletelyIntraShardRelayedTransaction(tx *transaction.ApiTransactionResult) (bool, error) {
 	isNotarized := tx.NotarizedAtSourceInMetaNonce > 0 && tx.NotarizedAtDestinationInMetaNonce > 0
 	if !isNotarized {
-		return false
+		return false, nil
 	}
 
 	isRelayedTransaction := tx.ProcessingTypeOnSource == relayedTransactionDescriptor && tx.ProcessingTypeOnDestination == relayedTransactionDescriptor
 	if !isRelayedTransaction {
-		return false
+		return false, nil
+	}
+
+	senderAddress, err := tp.pubKeyConverter.Decode(tx.Sender)
+	if err != nil {
+		return false, err
+	}
+	receiverAddress, err := tp.pubKeyConverter.Decode(tx.Receiver)
+	if err != nil {
+		return false, err
+	}
+
+	dataField := string(tx.Data)
+	if strings.Index(dataField, relayedTxV1DataMarker) != 0 {
+		return false, fmt.Errorf("wrong relayed v1 data marker position")
```

Comment:
Correct, `ProcessingTypeOn*` identifies Relayed V1, and the check is for Relayed V1 :ballot_box_with_check: